src/libratbag-data.c

```diff
@@ -499,15 +530,14 @@ ratbag_device_data_new_for_id(struct ratbag *ratbag, const struct input_id *id)
 
 		rc = xasprintf(&file, "%s/%s", datadir, files[n]->d_name);
 		if (rc == -1)
-			goto out;
-		if (file_data_matches(ratbag, file, id, &data))
-			goto out;
+			break;
```

Comment:
Keep the goto.

Reply:
may I ask why? :D we're not writing assembly